pallets/xst/src/lib.rs

```diff
@@ -859,6 +881,14 @@ impl<T: Config> Pallet<T> {
         }
     }
 
+    fn convert_timestamp_to_number(timestamp: T::Moment) -> u64 {
```

Comment:
You can add constant of type T::Moment to Config trait to avoid such conversion and make oracle rate lifetime configurable from runtime crate